### PR TITLE
[typescript-fetch] fix serialization/deserialization with inheritance

### LINKED ISSUE #3646: [BUG] [typescript-fetch] serialize/deserialze not working with inheritence
##### Description

Properties of super-classes are not serialized/deserialized.

##### openapi-generator version

4.1.0

##### OpenAPI declaration file content or url

```yaml
components:
  schemas:
    Type:
      type: string
      enum:
      - B
    A:
      type: object
      discriminator:
        propertyName: type
      required:
      - type
      properties:
        whatever-a:
          type: string
        type:
          $ref: "#/components/schemas/Type"
    B:
      allOf:
        - $ref: "#/components/schemas/A"
        - type: object
          discriminator:
            propertyName: type
          properties:
            whatever-b:
              type: string
```

##### Suggest a fix

in modelGeneric.mustache add the lines with comment:

```javascript
export function {{classname}}FromJSON(json: any): {{classname}} {
    {{#hasVars}}
    return {
        {{#parent}}...{{{parent}}}FromJSON(json),{{/parent}}  // deserialize inherited properties as well
        {{#additionalPropertiesType}}
            ...json,
        {{/additionalPropertiesType}}
```

and

```javascript
export function {{classname}}ToJSON(value?: {{classname}}): any {
    {{#hasVars}}
    if (value === undefined) {
        return undefined;
    }
    return {
        {{#parent}}...{{{parent}}}ToJSON(value),{{/parent}}     // serialize inherited properties as well
        {{#additionalPropertiesType}}
            ...value,
        {{/additionalPropertiesType}}
```

### PR DESCRIPTION
**Pull request by stephanpelikan**
Fixes #3646
Also includes basic support of discriminators.

### CODE REVIEW

**Comment by macjohnny** [comment]
thanks for your PR.
Please follow the steps in https://github.com/OpenAPITools/openapi-generator/blob/master/.github/PULL_REQUEST_TEMPLATE.md

**Comment by stephanpelikan** [comment]
I've run ./bin/typescript-fetch-petstore.sh which succeeded. I've also with my application.

@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @nicokoenig @topce @akehir : Could you please review the PR.

Thanks in advance :-)

**Comment by macjohnny** [comment]
> I've run ./bin/typescript-fetch-petstore.sh which succeeded. I've also with my application.

please commit the files

**Comment by stephanpelikan** [comment]
[typescript-fetch-petstore.sh.log](https://github.com/OpenAPITools/openapi-generator/files/3546566/typescript-fetch-petstore.sh.log)

**Comment by macjohnny** [comment]
@stephanpelikan please commit and push the generated files in `samples/client/petstore/typescript-fetch`

File: modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache [hunk 1]
```diff
@@ -79,7 +105,11 @@ export function {{classname}}ToJSON(value?: {{classname}}): any {
     if (value === undefined) {
         return undefined;
     }
+    if (value == null) {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache]:
```suggestion
    if (value === null) {
```

**Comment by stephanpelikan** [comment]
> please commit and push the generated files in `samples/client/petstore/typescript-fetch`

[typescript-fetch-petstore.output.zip](https://github.com/OpenAPITools/openapi-generator/files/3549513/typescript-fetch-petstore.output.zip)

**Pull request merged**